admissions/utils.py

```diff
@@ -407,11 +410,179 @@ def internal_group_applicant_data(internal_group):
+
+
+def get_applicants_who_will_be_accepted(admission):
+    return admission.applicants.filter(will_be_admitted=True)
+
+
+def get_applicant_interest_qs_with_offers(admission):
+    return ApplicantInterest.objects.filter(
+        applicant__admission=admission, position_to_be_offered__isnull=False
+    )
+
+
+def parse_applicant_interest_qs_to_gql_applicant_preview(applicant_interest_qs):
+    from .schema import ApplicantPreview
+
+    parsed_applicant_interests = []
+    for applicant_interest in applicant_interest_qs:
+        flattened_applicant_data = ApplicantPreview(
+            id=to_global_id("ApplicantNode", applicant_interest.applicant.id),
+            full_name=applicant_interest.applicant.get_full_name,
+            phone=applicant_interest.applicant.phone,
+            offered_internal_group_position_name=applicant_interest.position_to_be_offered.name,
+            applicant_priority="N/A",
+        )
+        parsed_applicant_interests.append(flattened_applicant_data)
+    return parsed_applicant_interests
+
+
+def parse_applicant_qs_to_gql_applicant_preview(applicant_qs):
+    from admissions.schema import ApplicantPreview
+
+    parsed_applicant_list = []
+    for applicant in applicant_qs:
+        priority = get_applicant_position_offer(applicant)
+        flattened_applicant_data = ApplicantPreview(
+            id=to_global_id("ApplicantNode", applicant.id),
+            full_name=applicant.get_full_name,
+            phone=applicant.phone,
+            offered_internal_group_position_name=priority.internal_group_position.name,
+            applicant_priority=priority.applicant_priority,
+        )
+        parsed_applicant_list.append(flattened_applicant_data)
+
+    return parsed_applicant_list
+
+
+def admission_applicant_preview(admission):
+    """
+    Gets all Applicant and ApplicantInterest instances that will be
+    offered a position in the provided admission and returns them
+    as a parsed list of graphene.ObjectType ApplicantPreview objects.
+
+    :param admission: An Admission model instance of the applicants we want to parse
+    :returns: A list of applicants who will be given an offer
+    """
+    applicants = get_applicants_who_will_be_accepted(admission)
+    # Then we expand this with applicants with interest.
+    parsed_applicants = parse_applicant_qs_to_gql_applicant_preview(applicants)
+
+    applicant_interests = get_applicant_interest_qs_with_offers(admission)
+    parsed_applicant_interests = parse_applicant_interest_qs_to_gql_applicant_preview(
+        applicant_interests
+    )
+
+    final_applicant_list = [*parsed_applicants, *parsed_applicant_interests]
+    # Can consider re-ordering by name here
+    return final_applicant_list
+
+
+def get_admission_final_applicant_qs(admission):
+    """
+    Retrieves all applicants who will be joining KSG, including those who have been offered
+    some other position.
+    """
```

Comment:
Specify that this returns aa merged queryset using all those with `will_be_admitted` and `Applicant` instances filtered through `ApplicantInterst`